test/protocol/catch_test.rb

```diff
@@ -5,22 +5,41 @@
 module DEBUGGER__
   class CatchTest < TestCase
     PROGRAM = <<~RUBY
-      1| module Foo
-      2|   class Bar
-      3|     def self.a
-      4|       raise 'foo'
-      5|     end
-      6|   end
-      7|   Bar.a
-      8|   bar = Bar.new
-      9| end
+     1| def foo
+     2|   a = 1
+     3|   raise "foo"
+     4| end
+     5|
+     6| foo
     RUBY
 
-    def test_catch_stops_when_the_runtime_error_raised
+    def test_set_exception_breakpoints_sets_exception_breakpoints
       run_protocol_scenario PROGRAM do
-        req_set_exception_breakpoints
+        req_set_exception_breakpoints([{ name: "RuntimeError" }])
         req_continue
-        assert_line_num 4
+        assert_line_num 3
+        req_terminate_debuggee
+      end
+    end
+
+    def test_set_exception_breakpoints_unsets_exception_breakpoints
+      run_protocol_scenario PROGRAM, cdp: false do
+        req_set_exception_breakpoints([{ name: "RuntimeError" }])
+        req_set_exception_breakpoints([])
+        req_continue
```

Comment:
Can we finish the program with `req_terminate_debuggee`?

I know that there is no problem in the only DAP case, but it may confuse other users when they write test cases for both protocol.

Reply:
This won't work because the debuggee would already have exited and the terminate request will cause borken pipe error:

```
  Failed to send request because of #<Errno::EPIPE: Broken pipe>.
/Users/st0012/projects/debug/test/support/assertions.rb:97:in `assert_block'
/Users/st0012/projects/debug/test/support/protocol_utils.rb:498:in `rescue in send_request'
/Users/st0012/projects/debug/test/support/protocol_utils.rb:487:in `send_request'
/Users/st0012/projects/debug/test/support/protocol_utils.rb:502:in `send_dap_request'
/Users/st0012/projects/debug/test/support/protocol_utils.rb:153:in `req_terminate_debuggee'
test/protocol/catch_test.rb:30:in `block in test_set_exception_breakpoints_unsets_exception_breakpoints'
     27:         req_set_exception_breakpoints([{ name: "RuntimeError" }])
     28:         req_set_exception_breakpoints([])
     29:         req_continue
  => 30:         req_terminate_debuggee
     31:       end
     32:     end
     33:
/Users/st0012/projects/debug/test/support/protocol_utils.rb:299:in `execute_dap_scenario'
/Users/st0012/projects/debug/test/support/protocol_utils.rb:50:in `run_protocol_scenario'
test/protocol/catch_test.rb:26:in `test_set_exception_breakpoints_unsets_exception_breakpoints'
```

Reply:
Oh, that's right. Thank you for explaining to me.